docs/docs/features/backlight.md

```diff
@@ -0,0 +1,117 @@
+---
+title: Backlight
+sidebar_label: Backlight
+---
+
+Backlight is a feature used to control array of LEDs, usually placed through or under switches. Unlike [RGB Underglow](underglow.md), backlight currently allows only one color per LED, also LEDs are not addressable, so you can't control individual LEDs.
+
+## Enabling Backlight
+
+To enable backlight on your board or shield, add the following lines to your `.conf` file of your user config directory as such:
+
+```
+CONFIG_PWM=y
+CONFIG_LED_PWM=y
+CONFIG_ZMK_BACKLIGHT=y
+```
+
+If your board or shield does not have backlight configured, refer to [Adding Backlight to a Board](#adding-backlight-to-a-board).
+
+## Configuring Backlight
+
+There are various Kconfig options used to configure the backlight feature. These can all be set in the `.conf` file.
+
+| Option                               | Description                                           | Default |
+| ------------------------------------ | ----------------------------------------------------- | ------- |
+| `CONFIG_ZMK_BACKLIGHT_BRT_STEP`      | Brightness step in percent                            | 20      |
+| `CONFIG_ZMK_BACKLIGHT_BRT_START`     | Default brightness in percent                         | 40      |
+| `CONFIG_ZMK_BACKLIGHT_ON_START`      | Default backlight state                               | y       |
+| `CONFIG_ZMK_BACKLIGHT_AUTO_OFF_IDLE` | Turn off backlight when keyboard goes into idle state | y       |
+| `CONFIG_ZMK_BACKLIGHT_AUTO_OFF_USB`  | Turn off backlight when USB is disconnected           | n       |
+
+## Adding Backlight to a Board
+
+Backlight is always added to a board, not a shield.
+If you have a shield with backlight, you must add a `boards/` directory within your shield folder to define the backlight individually for each board that supports the shield.
+Inside the `boards/` folder, you define a `<board>.overlay` for each different board.
```

Comment:
Thinking about this more, since we have the GPIO backed driver available that should be board agnostic, I wonder if we should be documenting and suggesting:

1. Add GPIO backlight nodes/compatibles to shields.
2. For tested boards used w/ the shield, add a `shield/foo/boards/bar.overlay` to override and add the board-specific PWM nodes.

Doing so will ensure a shield would have reasonable "fallback" backlight on some new board used w/ a given shield.

Thoughts?

Reply:
I think it's a good idea

Reply:
Added a note about that